### PR TITLE
post comment by review id

### PR DESCRIPTION
**Pull request by robbiemcdonough**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const app = require("../app");
 const seed = require("../db/seeds/seed");
 const db = require("../db/connection");
 const testData = require("../db/data/test-data/index");
-const { get } = require("../app");
+const { get, post } = require("../app");
```
Comment (emilyb93) [inline __tests__/app.test.js]: remove this line, its just rogue vs code imports, you dont need them

File: __tests__/app.test.js [hunk 2]
```diff
@@ -141,3 +141,41 @@ describe("/api/review/:review_id/comments", () => {
...
+    const REVIEW_ID = 2;
+    return request(app)
+      .post(`/api/reviews/${REVIEW_ID}/comments`)
+      .send({ username: "bainesface", body: "" })
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("comment body not found");
+      });
+  });
+});
```
Comment (emilyb93) [inline __tests__/app.test.js:208]: What about if we post a comment to a review that doesnt exist? (9999)
or a review_id that isnt a number?
what if we send a comment that is missing some keys?

File: models/model.js [hunk 3]
```diff
@@ -44,3 +44,25 @@ exports.fetchCommentsByReviewID = (review_id) => {
...
     });
 };
+
+exports.insertCommentsByReviewID = (newComment, review_id) => {
+  return db
+    .query("SELECT * FROM users WHERE username = $1", [newComment.username])
+    .then((array) => {
+      if (array.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "invalid username" });
+      } else {
```
Comment (emilyb93) [inline models/model.js]: Sql will throw an error if you try to insert a comment with an invalid username anyway, so you could just handle this in a single query with an error handler

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Some more testing please :) 
Great work so far, really impressed!

**Review by yalicee — APPROVED** [review]
Really good error handling ;) Well done!

**Pull request merged**